src/main/scala/fr/acinq/bitcoin/scalacompat/Crypto.scala

```diff
@@ -0,0 +1,226 @@
+   * A bitcoin private key.
+   * A private key is valid if it is not 0 and less than the secp256k1 curve order when interpreted as an integer (most significant byte first).
+   * The probability of choosing a 32-byte string uniformly at random which is an invalid private key is negligible, so this condition is not checked by default.
+   * However, if you receive a private key from an external, untrusted source, you should call `isValid()` before actually using it.
+   */
+  case class PrivateKey(priv: bitcoin.PrivateKey) {
+    val value: ByteVector32 = priv.value
+
+    def add(that: PrivateKey): PrivateKey = PrivateKey(this.priv plus that.priv)
+
+    def subtract(that: PrivateKey): PrivateKey = PrivateKey(this.priv minus that.priv)
+
+    def multiply(that: PrivateKey): PrivateKey = PrivateKey(this.priv times that.priv)
+
+    def +(that: PrivateKey): PrivateKey = add(that)
+
+    def -(that: PrivateKey): PrivateKey = subtract(that)
+
+    def *(that: PrivateKey): PrivateKey = multiply(that)
+
+    def isZero: Boolean = priv.value == bitcoin.ByteVector32.Zeroes
+
+    def isValid: Boolean = priv.isValid
+
+    def publicKey: PublicKey = PublicKey(priv.publicKey())
+
+    /**
+     * @param prefix Private key prefix
+     * @return the private key in Base58 (WIF) compressed format
+     */
+    def toBase58(prefix: Byte): String = priv.toBase58(prefix)
+
+    def toHex: String = priv.toHex
+
+    override def toString = priv.toString
+  }
+
+  object PrivateKey {
+    def apply(data: ByteVector): PrivateKey = PrivateKey(new bitcoin.PrivateKey(data.toArray))
+
+    /**
+     * @param data serialized private key in bitcoin format
+     * @return the de-serialized key
+     */
+    def fromBin(data: ByteVector): (PrivateKey, Boolean) = {
+      val compressed = data.length match {
+        case 32 => false
+        case 33 if data.last == 1.toByte => true
+      }
+      (PrivateKey(data), compressed)
+    }
+
+    def fromBase58(value: String, prefix: Byte): (PrivateKey, Boolean) = {
+      val p = bitcoin.PrivateKey.fromBase58(value, prefix)
+      (p.getFirst, p.getSecond)
+    }
+  }
+
+  /**
+   * A bitcoin public key (in compressed form).
+   * A public key is valid if it represents a point on the secp256k1 curve.
+   * The validity of this public key is not checked by default, because when you create a public key from a private key it will always be valid.
+   * However, if you receive a public key from an external, untrusted source, you should call `isValid()` before actually using it.
+   */
+  case class PublicKey(pub: bitcoin.PublicKey) {
+    val value: ByteVector = pub.value
+
+    def hash160: ByteVector = ByteVector.view(pub.hash160())
+
+    def isValid: Boolean = pub.isValid
+
+    def add(that: PublicKey): PublicKey = PublicKey(this.pub plus that.pub)
+
+    def add(that: PrivateKey): PublicKey = PublicKey(this.pub plus that.priv.publicKey())
+
+    def subtract(that: PublicKey): PublicKey = PublicKey(this.pub plus that.pub)
```

Comment:
It's weird, I think we had previously fixed that in earlier commits?

```suggestion
    def subtract(that: PublicKey): PublicKey = PublicKey(this.pub minus that.pub)
```

Reply:
I thought so too, fixed in https://github.com/ACINQ/bitcoin-lib/pull/63/commits/c67bb611f713af65c71ebea6354db1b9ca951a91